internal/cmd/controller/gitops/operator.go

```diff
@@ -138,21 +139,27 @@ func (g *GitOperator) Run(cmd *cobra.Command, args []string) error {
 		workers = w
 	}
 
+	bundleCreationMaxConcurrency, err := fleetapply.GetBundleCreationMaxConcurrency()
+	if err != nil {
+		setupLog.Error(err, "failed to parse FLEET_BUNDLE_CREATION_MAX_CONCURRENCY, using defaults", "env_var_name", fleetapply.BundleCreationMaxConcurrencyEnv)
+	}
+
 	kh := ssh.KnownHosts{EnforceHostKeyChecks: !g.SkipHostKeyChecks}
 
 	gitJobReconciler := &reconciler.GitJobReconciler{
-		Client:          mgr.GetClient(),
-		Scheme:          mgr.GetScheme(),
-		Image:           g.Image,
-		Scheduler:       sched,
-		Workers:         workers,
-		ShardID:         g.ShardID,
-		JobNodeSelector: g.ShardNodeSelector,
-		GitFetcher:      &git.Fetch{KnownHosts: kh},
-		Clock:           reconciler.RealClock{},
-		Recorder:        mgr.GetEventRecorderFor(fmt.Sprintf("fleet-gitops%s", shardIDSuffix)),
-		SystemNamespace: namespace,
-		KnownHosts:      kh,
+		Client:                       mgr.GetClient(),
+		Scheme:                       mgr.GetScheme(),
+		Image:                        g.Image,
+		Scheduler:                    sched,
+		Workers:                      workers,
+		ShardID:                      g.ShardID,
+		JobNodeSelector:              g.ShardNodeSelector,
+		GitFetcher:                   &git.Fetch{KnownHosts: kh},
+		Clock:                        reconciler.RealClock{},
+		Recorder:                     mgr.GetEventRecorderFor(fmt.Sprintf("fleet-gitops%s", shardIDSuffix)),
+		SystemNamespace:              namespace,
+		KnownHosts:                   kh,
+		BundleCreationMaxConcurrency: bundleCreationMaxConcurrency,
```

Comment:
Do we need the variable to be propagated in here?

As it's an env variable that only affects the `gitjob` creation we could just simple read it when calling `argsAndEnvs`, right? 

It is just to keep things simpler.